src/async.rs

```diff
@@ -365,8 +372,49 @@ impl<S: Sleeper> AsyncClient<S> {
     }
 
     /// Broadcast a [`Transaction`] to Esplora
-    pub async fn broadcast(&self, transaction: &Transaction) -> Result<(), Error> {
-        self.post_request_hex("/tx", transaction).await
+    pub async fn broadcast(&self, transaction: &Transaction) -> Result<Txid, Error> {
+        let body = serialize::<Transaction>(transaction).to_lower_hex_string();
+        let response = self.post_request_bytes("/tx", body, None).await?;
+        let txid = Txid::from_str(&response.text().await?).map_err(|_| Error::InvalidResponse)?;
+        Ok(txid)
+    }
```

Comment:
I just noticed that landing this PR has a breaking change, due to this change here and in the blocking client.

@tnull Do you need the submit_package API as a non-breaking release?